### PR TITLE
[WIP] Associate .wpost files with OLW

### PR DESCRIPTION
**Pull request by ScottIsAFool**

### CODE REVIEW

File: src/managed/OpenLiveWriter/ApplicationMain.cs [hunk 1]
```diff
@@ -190,6 +196,90 @@ private static bool LaunchAction(string[] args, bool isFirstInstance)
...
+
+            // Delete the key instead of trying to change it
+            var currentUser = Registry.CurrentUser.OpenSubKey($"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\FileExts\\{extension}", true);
+            currentUser?.DeleteSubKey("UserChoice", false);
+            currentUser?.Close();
+
+            // Tell explorer the file association has been changed
+            SHChangeNotify(0x08000000, 0x0000, IntPtr.Zero, IntPtr.Zero);
+        }
+
```
Comment (aviade) [inline src/managed/OpenLiveWriter/ApplicationMain.cs]: This should be encapsulated into a separate class.

File: src/managed/OpenLiveWriter/ApplicationMain.cs [hunk 2]
```diff
@@ -190,6 +196,90 @@ private static bool LaunchAction(string[] args, bool isFirstInstance)
...
+                        onInitialInstall: v => InitialInstall(mgr),
+                        onFirstRun: () => FirstRun(mgr),
+                        onAppUpdate: v => OnAppUpdate(mgr),
+                        onAppUninstall: v => OnAppUninstall(mgr));
+                }
+            }
+            catch (Exception)
+            {
+                // Ignore, chances are this is being run locally on a dev's machine
+            }
```
Comment (aviade) [inline src/managed/OpenLiveWriter/ApplicationMain.cs]: Use a DEBUG tag to support dev box. If this fails in release code the error should be logged.

**Comment by jongalloway** [comment]
Merged, thanks!

**Comment by Anthony2k8900** [comment]
Duplicate of # @dnfadmin for use. Use as follows 808be12 into openLive.msdn on Jan 2020 DEBUG coding

**Pull request merged**